work.py

```diff
@@ -360,6 +360,9 @@ def sync_project_contacts(cls, activities):
         Work = pool.get('project.work')
         to_save = []
         for activity in activities:
+            if not activity.resource:
+                continue
+
```

Comment:
I don't understand why we need this check. The case in which activity.resource is None is already covered by the `if isinstance` below.